### PR TITLE
Add table aws_oam_sink closes #1456

### LINKED ISSUE #1456: Add table aws_oam_sink
**References**
https://github.com/aws/aws-sdk-go-v2/blob/main/service/oam/api_op_ListSinks.go#L17

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  name,
  id,
  arn,
  tags,
  title
from
  aws_oam_sink;
+--------+--------------------------------------+------------------------------------------------------------------------------+-----------------------------+--------+
| name   | id                                   | arn                                                                          | tags                        | title  |
+--------+--------------------------------------+------------------------------------------------------------------------------+-----------------------------+--------+
| test43 | c7bf4c81-7bdf-4cbd-a4b5-5dfc61b1489f | arn:aws:oam:us-east-1:934502159546:sink/c7bf4c81-7bdf-4cbd-a4b5-5dfc61b1489f | {"foo":"bar","foo1":"bar1"} | test43 |
+--------+--------------------------------------+------------------------------------------------------------------------------+-----------------------------+--------+
```
</details>

### CODE REVIEW

File: aws/plugin.go [hunk 1]
```diff
@@ -281,6 +281,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_neptune_db_cluster":                                       tableAwsNeptuneDBCluster(ctx),
 			"aws_networkfirewall_firewall_policy":                          tableAwsNetworkFirewallPolicy(ctx),
 			"aws_networkfirewall_rule_group":                               tableAwsNetworkFirewallRuleGroup(ctx),
+			"aws_oam_sink":                                                 tableAwsOAMSink(ctx),
```
Comment (misraved) [inline aws/plugin.go:284]:
```suggestion
			"aws_oam_sink":                                                 tableAwsOamSink(ctx),
```

File: aws/table_aws_oam_sink.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/oam"
+	"github.com/aws/aws-sdk-go-v2/service/oam/types"
+)
+
+func tableAwsOAMSink(_ context.Context) *plugin.Table {
```
Comment (misraved) [inline aws/table_aws_oam_sink.go:15]:
```suggestion
func tableAwsOamSink(_ context.Context) *plugin.Table {
```

File: aws/table_aws_oam_sink.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+func listAwsOamSinks(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create client
+	svc, err := OAMClient(ctx, d)
```
Comment (misraved) [inline aws/table_aws_oam_sink.go:73]: Don't we need the unsupported region check for this API call?

File: aws/table_aws_oam_sink.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+	if resp != nil {
+		return resp.Tags, nil
+	}
+
+	return nil, nil
+}
+
+//// UTILITY FUNCTION
+
+func getSinkArn(item interface{}) string {
```
Comment (misraved) [inline aws/table_aws_oam_sink.go:187]: Should this function be renamed to `getOamSinkArn`?

File: docs/tables/aws_oam_sink.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  name,
+  id,
+  arn,
+  tags,
+  title
+from
+  aws_oam_sink;
+```
+
+### Get sink by id
```
Comment (misraved) [inline docs/tables/aws_oam_sink.md:20]:
```suggestion
### Get sink by ID
```

File: aws/table_aws_oam_sink.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+			/// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
```
Comment (misraved) [inline aws/table_aws_oam_sink.go:55]: Don't we need the `tag_src` column?

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments/questions. Thanks!!!

File: aws/table_aws_oam_sink.go (hunk 6 above)
Comment (ParthaI) [reply aws/table_aws_oam_sink.go:55]: We don't need `tag_src` column as we are getting response in `map[string]string` type.

**Review by misraved — APPROVED** [review]

**Pull request merged**